qemu/tests/cgroup.py

```diff
@@ -1133,9 +1133,11 @@ def _generate_verification(cpusets, cpus):
             logging.error("Test requires at least 2 vCPUs.")
             raise exceptions.TestSkipError("Test requires at least 2 vCPUs.")
         # Check whether smp changed and recreate VM if so
-        if vm_cpus != params.get("smp", 0):
+        if vm_cpus != (params.get_numeric("smp") or params.get_numeric("vcpu_maxcpus")):
             logging.info("Expected VM reload.")
             params['smp'] = vm_cpus
+            if params.get("cgroup_use_half_smp") == "yes":
+                params['vcpu_maxcpus'] = params['smp']
```

Comment:
Why when `cgroup_use_half_smp == yes`, `vcpu_maxcpus` will equal to `smp`.

Reply:
The guest needs an half number of host's cpu as the test case requires, if we don't set the same, the vcpu_maxcpus is less than smp. The creation fails. Thanks.

Reply:
We should make vcpu_maxcpus the same as smp, but I can not get the mean of cgroup_use_half_smp, your changes don't have any host CPU setting.

Reply:
will update it to "params['vcpu_maxcpus'] = params['smp'] = vm_cpus"